### PR TITLE
GEOS-8509 Basic geofence plugin docs

### PR DESCRIPTION
**Pull request by NielsCharlier**

### CODE REVIEW

File: doc/en/user/source/community/geofence/cache.rst [hunk 1]
```diff
@@ -0,0 +1,48 @@
+.. _geofence_cache:
+
+GeoFence Cache REST
+===================
+
+The Geofence client cache can be queried and invalidated through a REST service.
```
Comment (bradh) [inline doc/en/user/source/community/geofence/cache.rst]: We aren't really querying the cache, more querying the cache status. Also, invalidated isn't very user-speak, so perhaps something like: "The Geofence client cache status can be queried, and the cache cleared (invalidated) through a REST service."

File: doc/en/user/source/community/geofence/cache.rst [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+Requests
+--------
+
+``/geofence/ruleCache/info``
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Retrieve information about the geofence caches.
```
Comment (bradh) [inline doc/en/user/source/community/geofence/cache.rst]: cache singular?

File: doc/en/user/source/community/geofence/cache.rst [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+.. list-table::
+   :header-rows: 1
+   :widths: 10 20 20 20
+
+   * - Method
+     - Action
+     - Parameters
+     - Response
+   * - GET
+     - Retrieve information about the geofence caches  (such as cache size, hits, misses, success, failures, load times and evictions)
```
Comment (bradh) [inline doc/en/user/source/community/geofence/cache.rst]: cache singular? What is the difference between hits/misses and success/failures ? If you want to keep them all, it probably needs to be "successes" to match the other plural forms.

File: doc/en/user/source/community/geofence/cache.rst [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+     - ---
+     - 200 OK. Text Format.
+
+
+``/geofence/ruleCache/invalidate``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Invalidate the geofence cache.
```
Comment (bradh) [inline doc/en/user/source/community/geofence/cache.rst]: maybe "Invalidate (clear) the geofence cache." here too.

File: doc/en/user/source/community/geofence/configuration.rst [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+The GeoFence Admin Page is a component of the GeoServer web interface. You can access it from the GeoServer web interface by clicking the :guilabel:`GeoFence` link, found on the left side of the screen after logging in.
+
+.. figure:: images/configuration.png
+   :align: center
+
+General Settings
+----------------
+Configure the following settings here:
+
+- Geoserver instance name: the name under which this geoserver is known by the geofence server. This useful for when you use an external geofence server with multiple geoserver server.
```
Comment (bradh) [inline doc/en/user/source/community/geofence/configuration.rst]: perhaps "known to the geofence server" at the end of the first sentence. "multiple geoserver server**s**." (plural) at the end of the second sentence.

File: doc/en/user/source/community/geofence/configuration.rst [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+- GeoServer services URL: this is how geoserver knows how to connect to the external geofence server. When using an internal geofence geoserver, this is not configurable.
+
```
Comment (bradh) [inline doc/en/user/source/community/geofence/configuration.rst]: The "internal geofence geoserver" probably should be "internal geofence server" or "internal geofrence service". Maybe could use an example here, or a reference to some other docs, for the URL.

File: doc/en/user/source/community/geofence/configuration.rst [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+Configure the following settings here:
+
+- Allow remote and inline layers in SLD
+
+- Authenticated users can write
+
+- Use GeoServer roles to get authorizations
+
+- Comma delimited list of mutually exclusive roles for authorization
+
```
Comment (bradh) [inline doc/en/user/source/community/geofence/configuration.rst]: These four settings (especially the last two) probably need some more explanation - perhaps an outreference? Not a blocker for this.

File: doc/en/user/source/community/geofence/configuration.rst [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+Cache
+-----
+
+Configure the following settings here:
+
+- Size of the rule cache
```
Comment (bradh) [inline doc/en/user/source/community/geofence/configuration.rst]: Needs units.

File: doc/en/user/source/community/geofence/configuration.rst [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+- Cache refresh interval (ms)
+
+- Cache expire interval (ms)
+
+Collected data about the caches can be retrieved here (such as cache size, hits, misses, success, failures, load times and evictions) and the cache can be manually invalidated.
```
Comment (bradh) [inline doc/en/user/source/community/geofence/configuration.rst]: Again, add "(cleared)" to the end.

File: doc/en/user/source/community/geofence/index.rst [hunk 10]
```diff
@@ -0,0 +1,12 @@
+.. _community_geofence:
+
+Geofence Plugin
+===============
+`GeoFence <https://github.com/geoserver/geofence/>`_ offers an alternative to the GeoServer :ref:`security` substem of GeoServer, allowing far more advanced security configurations, such as rules that combine data and service restrictions. It uses a client-server model, and this plugin only provides the client component. It must connect either to an external Geofence server, or be used in combination with the GeoServer integrated Geofence server  :ref:`community_geofence_server`.
```
Comment (bradh) [inline doc/en/user/source/community/geofence/index.rst]: substem -> subsystem.

File: doc/en/user/source/community/geofence/installing.rst [hunk 11]
```diff
@@ -0,0 +1,12 @@
+.. _geofence_install:
+
+Installing the GeoServer GeoFence extension
+===========================================
+
+ #. Download the extension from the `nightly GeoServer community module builds <http://ares.opengeo.org/geoserver/master/community-latest/>`_.
```
Comment (bradh) [inline doc/en/user/source/community/geofence/installing.rst]: Are we still on Ares for this?

File: doc/en/user/source/community/geofence/installing.rst [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+
+    .. warning:: Make sure to match the version of the extension to the version of the GeoServer instance!
```
Comment (bradh) [inline doc/en/user/source/community/geofence/installing.rst]: "Be sure to" or "Make sure you" would be better.

File: doc/en/user/source/community/geofence/cache.rst (hunk 2 above)
Comment (NielsCharlier) [reply doc/en/user/source/community/geofence/cache.rst]: there are three caches: rule cache, admin cache and user cache

File: doc/en/user/source/community/geofence/cache.rst (hunk 3 above)
Comment (NielsCharlier) [reply doc/en/user/source/community/geofence/cache.rst]: I am assuming that cache hits and misses refers to finding the object in the cache, or not. (load) successes and failures refers to loading the object when there is a cache miss. I'll add the word "load" to successes and failures

File: doc/en/user/source/community/geofence/configuration.rst (hunk 8 above)
Comment (NielsCharlier) [reply doc/en/user/source/community/geofence/configuration.rst]: yeah it is odd they are not there in the GUI while the other ones do have units

Comment (NielsCharlier) [reply doc/en/user/source/community/geofence/configuration.rst]: code tells me it's the amount of entries

Comment (NielsCharlier) [reply doc/en/user/source/community/geofence/configuration.rst]: code tells me it's number of entries

File: doc/en/user/source/community/geofence/configuration.rst (hunk 7 above)
Comment (NielsCharlier) [reply doc/en/user/source/community/geofence/configuration.rst]: the problem is that I am not so sure myself. Perhaps someone who has better understanding of these settings can add text or clarify here? (@mbarto ?)

**Comment by NielsCharlier** [comment]
Changes were made as requested

File: doc/en/user/source/community/geofence/cache.rst [hunk 13]
```diff
@@ -3,15 +3,15 @@
...
+The Geofence client cache status can be queried, and the cache cleared (invalidated) through a REST service.
 
 Requests
 --------
 
 ``/geofence/ruleCache/info``
 ~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Retrieve information about the geofence caches.
+Retrieve information about the geofence cache statys.
```
Comment (bradh) [inline doc/en/user/source/community/geofence/cache.rst]: statys => status.

**Pull request merged**